### PR TITLE
SLING-10693 fail build in case of invalid extends/requires file

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java [hunk 1]
```diff
@@ -118,81 +123,86 @@ void processRequiresFile(@NotNull Path requiresFile,
...
-            Path scriptFile = scriptPath.getFileName();
-            if (scriptFile != null) {
-                Path relativeResourceTypeFolder = resourceTypeDirectory.relativize(scriptPath);
-                int pathSegments = relativeResourceTypeFolder.getNameCount();
-                LinkedHashSet<String> selectors = new LinkedHashSet<>();
-                if (pathSegments > 1) {
-                    for (int i = 0; i < pathSegments - 1; i++) {
-                        selectors.add(relativeResourceTypeFolder.getName(i).toString());
-                    }
+        Path scriptFile = scriptPath.getFileName();
```
Comment (raducotescu) [inline src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java:151]: What was the point of refactoring this chunk? It doesn't seem to be related to the issue.

File: pom.xml [hunk 2]
```diff
@@ -49,6 +49,7 @@
         <maven.version>3.6.0</maven.version>
         <site.jira.version.id>12344664,12347157,12348422,12350010</site.jira.version.id>
         <maven.site.path>${project.artifactId}-archives/${project.artifactId}-LATEST</maven.site.path>
+        <project.build.outputTimestamp>2020-01-22T15:10:38Z</project.build.outputTimestamp>
```
[the remark this answers is not in the document]
Comment (kwin) [reply pom.xml:52]: This is related to the parent update and https://maven.apache.org/guides/mini/guide-reproducible-builds.html

File: src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java (hunk 1 above)
Comment (kwin) [reply src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java:151]: Initially I thought that we should also throw an exception if no engine was found for an extension. This was too strict though, but it is important to always log a WARN to make users aware of potential misconfigurations. I only removed the condition of line 140 because of this. I am gonna restore the condition for the empty extension to make it more visible.

Comment (kwin) [reply src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java]: Done in https://github.com/apache/sling-scriptingbundle-maven-plugin/pull/4/commits/4d6b9c5c26e17f21c5ef2b40072423f31151dc53

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (raducotescu) [reply pom.xml:52]: Then let's handle this in a separate issue, please. You don't need to update the parent POM for SLING-10693, right?

File: src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java (hunk 1 above)
Comment (raducotescu) [reply src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java]: Thanks! Looks better now.

**Review by raducotescu — APPROVED** [review]
I'm ok with merging the current changes, except for the POM update, which IMO warrants an issue on its own.

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kwin) [reply pom.xml]: I reverted those unrelated changes in https://github.com/apache/sling-scriptingbundle-maven-plugin/pull/4/commits/83960ac0cec00af6faacd6616b3f949c2b87c140.
